exopy_qm/tasks/tasks/ResumeAndGetDataTask.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import numpy as np
+from inspect import cleandoc
+import time
+import qm.qua
+from atom.api import Float, Int, List, Typed, Str, Value, Bool, set_default
+from exopy.tasks.api import InstrumentTask
+
+logger = logging.getLogger(__name__)
+
+
+class ResumeAndGetDataTask(InstrumentTask):
```

Comment:
I don't really understand the difference with `MeasureWithPauseTask`. Is this just an upgrade?

Reply:
If that is the case, let's delete the old task to avoid confusion

Reply:
The difference is that data is fetched only once, outside of the external loop

Reply:
That's may be caused by the way you write the QUA but both tasks boil down to resume + fetch_all so if you use save_all in the QUA program, you can also get all the data at once.

Reply:
![image](https://github.com/Exopy/exopy_qm/assets/57400235/d4de51fc-4765-4024-9da8-6cc8b5e48bdd)

It would look like this. 

I also did this because the wait_for_all_results that does not work if the program is paused. So I wanted a way to separate the "resume" task from the "get data" task, and to terminate the program before fetching all the data.

The iterate/finish uses input stream to tell to continue or finish, the resume/ResumAndGetData need to input the number of repetitions in the program.

All this to deal with the problem that in the current measure with pause setting, there is no token that tells that the last result is ready to be fetched. This token is integrated in the wait_for_all_results that works only if the program is terminated.

Reply:
![image](https://github.com/Exopy/exopy_qm/assets/57400235/13cd824f-61c3-457a-b4db-0692e3f67917)

(Or like this)

Reply:
Oh, I see. This is quite confusing. Feeding this input stream is also not guaranteed to work because the streams are not necessarily synchronised (one can be ready but not the other). The input stream  "iterate" could be ready before the actual data.
I still don't see the difference with MeasureWithPauseTask since the finish part of this setup is handled in a separate task. Both tasks are still resume-fetch_all-save data (confusingly, the FinishProgramTask also saves the data???)
If you are using a `save_all` though (as in this example), a hardcoded waiting time at the end would be enough.

Reply:
This is not a problem if the data is not ready, the stream is being processed asynchronously.
The difference is that iterate does not fetch any data, it only iterates it (as before in the infinite_loop_(), but now it should be a while_(iterate)). The save_all occurs at the end of the loop.

I realize that my first picture is confusing, the config task should be in the loop0:

![image](https://github.com/Exopy/exopy_qm/assets/57400235/4dec3229-a030-45f4-86d1-4609ba7222b0)

Reply:
I admit that the names are not well chosen though... I did not want to have too long names.